rust/impls/src/in_memory_store.rs

```diff
@@ -0,0 +1,350 @@
+use crate::{VssDbRecord, LIST_KEY_VERSIONS_MAX_PAGE_SIZE, MAX_PUT_REQUEST_ITEM_COUNT};
+use api::error::VssError;
+use api::kv_store::{KvStore, GLOBAL_VERSION_KEY, INITIAL_RECORD_VERSION};
+use api::types::{
+	DeleteObjectRequest, DeleteObjectResponse, GetObjectRequest, GetObjectResponse, KeyValue,
+	ListKeyVersionsRequest, ListKeyVersionsResponse, PutObjectRequest, PutObjectResponse,
+};
+use async_trait::async_trait;
+use bytes::Bytes;
+use chrono::prelude::Utc;
+use std::collections::BTreeMap;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+fn build_storage_key(user_token: &str, store_id: &str, key: &str) -> String {
+	format!("{}#{}#{}", user_token, store_id, key)
+}
+
+/// In-memory implementation of the VSS Store.
+pub struct InMemoryBackendImpl {
+	store: Arc<Mutex<BTreeMap<String, VssDbRecord>>>,
```

Comment:
No need to hold this in an `Arc` AFAICT.
```suggestion
	store: Mutex<BTreeMap<String, VssDbRecord>>,
```